design/ceph/ceph-nvmeof-gateway.md

```diff
@@ -0,0 +1,210 @@
+                                           +---------|-----------+
+                                                     |
+                                      +--------------v--------------+
+                                      |        Client Node          |
+                                      +-----------------------------+
+                                      | NVMe-oF Initiator (nvme-cli)|
+                                      |                             |
+                                      |  Application sees           |
+                                      |  /dev/nvmeXnY               |
+                                      +-----------------------------+
+
+
+
+
+
+* Note:  In this document, when we refer to 'NVMe namespace,' we mean the NVMe protocol's concept of a logical block device, not Kubernetes resource namespaces."
+
+### Components Responsibilities
+
+* **Volume Provisioning**: CSI driver creates RBD image and configures it in NVMe-oF gateway.
+* **Gateway Management**: Rook operator manages gateway lifecycle, scaling, and health.
+* **Load Balancing/k8s Service**: Service provides single endpoint for multiple gateway instances, NVMe namespace exposed over TCP/IP fabric with port 5500.
+* **Client Access**: External clients connect via NVMe-oF initiator to access volumes.
+
+## CephNVMeOFGateway CRD
+
+### Gateway Configuration
+
+- **Group**: `gateway-group-1` - Logical grouping of gateways
+- **Replicas**: `2` - Two gateway instances for HA
+- **Port**: `5500` - NVMe-oF service port
+
+
+### Example CephNVMeOFGateway Resource
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephNVMeOFGateway
+metadata:
+  name: nvmeof-gateway
+  namespace: rook-ceph
+spec:
+  gatewayGroup: "gateway-group-1"
+  replicas: 2
+  port: 5500
+  grpcPort: 5501
+  placement:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: role
+            operator: In
+            values: ["osd-node"]
+  resources:
+    #  limits:
+    #    cpu: "500m"
+    #    memory: "1024Mi"
+    #  requests:
+    #    cpu: "500m"
+    #    memory: "1024Mi"
+apiVersion: ceph.rook.io/v1
+kind: CephNVMeOFGateway
+metadata:
+  name: nvmeof-gateway
+  namespace: rook-ceph
+spec:
+  gatewayGroup: "gateway-group-1"
+  replicas: 2
+  port: 5500
+  
+  placement:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: role
+            operator: In
+            values: ["osd-node"]
+  
+  resources:
+    requests:
+      cpu: "500m"
+      memory: "1024Mi"
+  # Keep subsystems in CRD - Rook creates them via gRPC API
+  subsystems:
+  - nqn: "nqn.2016-06.io.spdk:production"
+    # hosts: # Initially empty, CSI will add hosts dynamically on PVC creation
+```
+
+Note: As per Ceph documentation, NVMe-oF gateways, which can either be colocated with OSD nodes or on dedicated nodes.
```

Comment:
If it's just about ensuring sufficient resources, that is true for any daemon, so I would suggest to remove this note. Or Is there a special consideration for co-location that I'm missing?